### PR TITLE
Draft: Sandbox/Notes from trying to replicate do_eks in lincbrain ecosystem

### PR DESCRIPTION
**Pull request by aaronkanzer**
Cc @satra @kabilar @asmacdo -- these are some of the notes/updates I have had to do locally to try to replicate the `do_eks` work

Still tinkering around -- the first `install.sh` call was pretty successful (albeit karpenter struggled to find the correct networking setup to assign pods and spin up nodes

cleanup.sh worked well on first try too; however, I think we will need to identify a checklist of remaining artifacts that one would need to delete in AWS in order to have a clean re-install.

Can elaborate more when we meet -- thanks

### CODE REVIEW

File: dandihub.tfvars [hunk 1]
```diff
@@ -1,6 +1,10 @@
 singleuser_image_repo = "dandiarchive/dandihub"
 region = "us-west-1"
 singleuser_image_tag = "latest"
-jupyterhub_domain = "hub-staging.dandiarchive.org"
-dandi_api_domain = "https://api.dandiarchive.org"
-admin_users = ["asmacdo", "dandibot", "satra", "yarikoptic"]
+jupyterhub_domain = "hub-sandbox.lincbrain.org"
+dandi_api_domain = "https://staging-api.lincbrain.org"
+admin_users = ["asmacdo", "dandibot", "satra", "aaronkanzer", "kabilar"]
```
Comment (asmacdo) [inline dandihub.tfvars:6]: We shouldn't have to touch the dandihub.tfvars for linc

File: install.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ source ensure_vars.sh
 echo "Initializing ..."
 terraform init || echo "\"terraform init\" failed"
 
-varfile="$HUB_DEPLOYMENT_NAME.tfvars"
```
Comment (asmacdo) [inline install.sh:12]: Did this not work as expected?

File: sandbox-lincbrain.tfvars [hunk 3]
```diff
@@ -0,0 +1,11 @@
+singleuser_image_repo = "dandiarchive/dandihub"
+region = "us-west-1"
+singleuser_image_tag = "latest"
+jupyterhub_domain = "hub-sandbox.lincbrain.org"
+jupyterhub_name = "sandbox-lincbrain"
+dandi_api_domain = "https://staging-api.lincbrain.org"
+admin_users = ["asmacdo", "dandibot", "satra", "aaronkanzer", "kabilar"]
+github_client_id = "fffb5efe40b01617b837"
+github_client_secret = "<something-i-am-storing-locally>"
```
Comment (asmacdo) [inline sandbox-lincbrain.tfvars:9]: this should be set in an env.var rather than here

File: sandbox-lincbrain.tfvars [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+danditoken = "0ead6c0e8fad218d9beb795c2889bc597a0220a8"
```
Comment (asmacdo) [inline sandbox-lincbrain.tfvars]: this should also be env var i think

File: ensure_vars.sh [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+while IFS= read -r line; do
+    if [[ $line =~ ^[^#]*= ]]; then
+        key=$(echo $line | cut -d '=' -f 1 | xargs)
+        value=$(echo $line | cut -d '=' -f 2- | xargs)
+        declare TF_VAR_$key="$value"
+        export TF_VAR_$key
+    fi
+done < "sandbox-lincbrain.tfvars"
+
+export HUB_DEPLOYMENT_NAME="sandbox-lincbrain"
```
Comment (aaronkanzer) [inline ensure_vars.sh:12]: I added this blurb to handle the env var situation

File: sandbox-lincbrain.tfvars (hunk 4 above)
Comment (aaronkanzer) [reply sandbox-lincbrain.tfvars]: I added this just to resolve.

https://github.com/lincbrain/linc-hub/pull/1/files#r1561742254

File: sandbox-lincbrain.tfvars (hunk 3 above)
Comment (aaronkanzer) [reply sandbox-lincbrain.tfvars:9]: https://github.com/lincbrain/linc-hub/pull/1#discussion_r1561743653

File: sandbox-lincbrain.tfvars (hunk 4 above)
Comment (aaronkanzer) [reply sandbox-lincbrain.tfvars]: Just to note for security sake: I renewed the token as well FYI, so that one is expired

**Comment by kabilar** [comment]
Closing since we are merging all LINC Hub configurations upstream.

**Pull request closed without merging**